operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/DefaultEventHandler.java

```diff
@@ -146,7 +124,13 @@ private void executeBufferedEvents(String customResourceUid) {
               latestCustomResource.get(),
               retryInfo(customResourceUid));
       log.debug("Executing events for custom resource. Scope: {}", executionScope);
-      executor.execute(new ExecutionConsumer(executionScope, eventDispatcher, this));
+      executor.execute(() -> {
+        // change thread name for easier debugging
+        Thread.currentThread().setName("EventHandler-" + controllerName);
```

Comment:
Is this needed? It looks a bit strange that we change the thread name on every execution.

Reply:
Well, that was previously done when initializing the thread pool by specifying the thread factory. We can't do that anymore so the only alternative I could think of at that point was to rename the thread when it executes.